python/pyspark/sql/streaming.py

```diff
@@ -0,0 +1,124 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from abc import ABCMeta, abstractmethod
+
+from pyspark import since
+
+__all__ = ["ContinuousQuery", "ProcessingTime"]
+
+
+class ContinuousQuery(object):
+    """
+    A handle to a query that is executing continuously in the background as new data arrives.
+    All these methods are thread-safe.
+
+    .. note:: Experimental
+
+    .. versionadded:: 2.0
+    """
+
+    def __init__(self, jcq, sqlContext):
+        self._jcq = jcq
+        self._sqlContext = sqlContext
+
+    @property
+    @since(2.0)
+    def name(self):
+        """The name of the continuous query.
+        """
+        return self._jcq.name()
+
+    @property
+    @since(2.0)
+    def isActive(self):
+        """Whether this continuous query is currently active or not.
+        """
+        return self._jcq.isActive()
+
+    @since(2.0)
+    def awaitTermination(self):
```

Comment:
One method returns Unit, the other returns something. I wasn't sure if I
should keep that semantic on the Python side
On Apr 14, 2016 12:27 AM, "Davies Liu" <notifications@github.com> wrote:

> In python/pyspark/sql/streaming.py
> <https://github.com/apache/spark/pull/12320#discussion_r59672622>:
>
> > +    @property
> > +    @since(2.0)
> > +    def name(self):
> > +        """The name of the continuous query.
> > +        """
> > +        return self._jcq.name()
> > +
> > +    @property
> > +    @since(2.0)
> > +    def isActive(self):
> > +        """Whether this continuous query is currently active or not.
> > +        """
> > +        return self._jcq.isActive()
> > +
> > +    @since(2.0)
> > +    def awaitTermination(self):
>
> this should be
>
> def awaitTermination(self, timeoutMs=None):
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/12320/files/1fe20edd78cee9851c2a973b543171a22021cdf9#r59672622>
>